homeassistant/components/garages_amsterdam/__init__.py

```diff
@@ -26,7 +26,8 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload Garages Amsterdam config entry."""
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
-    if len(hass.config_entries.async_entries(DOMAIN)) == 1:
+
+    if unload_ok:
         hass.data.pop(DOMAIN)
```

Comment:
This will pop the full domain now, as soon as multiple garages are set up and only 1 is unloaded?

Reply:
Maybe I should revert it, or change to something like this?

```python
if unload_ok:
    hass.data[DOMAIN].pop(entry.entry_id)
    if not hass.data[DOMAIN]:
        hass.data.pop(DOMAIN)
return unload_ok
```

Reply:
I don't think the above suggestion will work either, as there is only a single coordinator instance for the whole integration?

Reply:
Then I will revert to what it was